plugins/async-node/ios/Tests/AsynNodePluginTests.swift

```diff
@@ -292,6 +284,166 @@ class AsyncNodePluginTests: XCTestCase {
         XCTAssert(count == 3)
         XCTAssertEqual(expectedMultiNode2Text, "new node from the hook 2")
     }
+
+    func testAsyncNodeWithNullNode() {
+        let handlerExpectation = XCTestExpectation(description: "null node handled")
+
+        let context = JSContext()
+
+        let resolveHandler: AsyncHookHandler = { node in
+            handlerExpectation.fulfill()
+            XCTAssertNil(node)
+            return .singleNode(.concrete(JSValue()))
```

Comment:
This is testing the `.singleNode` case, right? But with an empty `JSValue`? We'd want to test the new enum case here, such that the addition would suffice towards resolving with "empty" (`nil` or `null` or `undefined` - these are really all different representations of empty, which we concretely tie back to a specific enum case).

Reply:
This actually lies inline to above comment that I was pretty thinking in the lines that be it (nil or null or undefined) these are already being handled , so I have tested it with the existing singlenode